src/types.rs

```diff
@@ -72,6 +72,8 @@ impl Clone for SeaRc<dyn Iden> {
     }
 }
 
+impl Eq for SeaRc<dyn Iden> {}
```

Comment:
Hmmmm... `Eq` for `dyn Iden`

Consider this. Where both `dyn Iden` yield the same identifier `id` but it's not the same Rust structure / enum.

```rs
#[derive(Iden)]
struct Id;

// Both yield the string "id"
assert_eq!(Id.to_string(), "id");
assert_eq!(Alias::new("id").to_string(), "id");

// How about this??
assert_eq!(Id, Alias::new("id"));
```

Reply:
Isn't this already covered by the `PartialEq` implementation that already exists for `DynIden`?  The current implementation checks two things: whether the .to_string() contents are equal and whether the vtable pointer is the same for the two objects, though this only works for dyn objects.

Anyway, if we wanted to provide the ability to do something like the last line in your example, it'd be necessary to implement something like `fn get_iden_id()` to allow users direct comparison between Iden objects, i.e. `Id.get_iden_id() == Alias::new("id").get_iden_id()` (which is probably already doable using `to_string()`). 

Though I must say, this feels rather unnatural. I don't think comparison between two different struct types is something a lot of users are doing. There are no feature requests for this either, so I think we're good in that regard.

Reply:
@tyt2y3 thoughts?